runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -402,14 +401,13 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 	/*MMINLINE*/ void
 	memcpyFromArray(void *destData, J9IndexableObject *srcObject, I_32 elementIndex, I_32 elementCount)
 	{
-		J9ROMArrayClass *romArrayClass = (J9ROMArrayClass*)J9GC_J9OBJECT_CLAZZ(srcObject)->romClass;
-		UDATA elementScale = romArrayClass->arrayShape & 0x0000FFFF;
+		UDATA elementSize = J9ARRAYCLASS_GET_STRIDE(J9GC_J9OBJECT_CLAZZ(srcObject));
 		if (isInlineContiguousArraylet(srcObject)) {
 			// If the data is stored contiguously, then a simple copy is sufficient.
 			void* srcData = getDataPointerForContiguous(srcObject);
-			switch(elementScale)
+			switch(elementSize)
```

Comment:
you should add `default:` case to all these switches to handle array sizes other then 1,2,4,8. You can add your own implementation (for example using `memcpy()`) or modify `case 1:`  to do proper byte by byte copy:
```
			case 1:
                        default:
					// byte/boolean or flattened type
					{
						U_8* srcCursor = ((U_8*)srcData) + elementIndex * elementSize;
					U_8* destCursor = (U_8*)destData;
					I_32 count = elementCount  * elementSize;
					while(count--) {
						*destCursor++ = *srcCursor++;
					}
					}
					break;
```

Reply:
Will there ever be a case where to elementsize is not 1, 2, 4, 8?
References have size 4, and the biggest primitive is a double which is byte size 8...

Reply:
@dmitripivkine I've also added default cases for the discontiguous case of `memcpyFromArray` as well as `memcpyToArray` an its discontiguous case. Could you review my changes to see if I've done this correctly?